scipy/signal/tests/test_signaltools.py

```diff
@@ -652,7 +652,24 @@ def test_basic(self):
         # make sure interpolated values are on knot points
         assert_array_almost_equal(y2[::10], y, decimal=5)
 
-
+    def test_complex(self):
+        # create some smoothly varying complex signal to try and interpolate
+        x = np.arange(100)
+        y = np.zeros(x.shape, dtype=np.complex64)
+        T = 10.0
+        f = 1.0 / T
+        y = np.sin(2.0 * np.pi * f * x) + \
```

Comment:
Also, yes please cut down on the number of points. For a big library like this, cutting test time is important.